src/spv/spv_rpc.cpp

```diff
@@ -445,13 +447,14 @@ UniValue spv_listanchors(const JSONRPCRequest& request)
     if (!spv::pspv)
         throw JSONRPCError(RPC_INVALID_REQUEST, "spv module disabled");
 
-    RPCTypeCheck(request.params, { UniValue::VNUM, UniValue::VNUM, UniValue::VNUM, UniValue::VNUM }, true);
+    RPCTypeCheck(request.params, { UniValue::VNUM, UniValue::VNUM, UniValue::VNUM, UniValue::VNUM, UniValue::VNUM, UniValue::VNUM }, true);
 
-
-    int minBtcHeight = request.params.size() > 0 && !request.params[0].isNull() ? request.params[0].get_int() : -1;
-    int maxBtcHeight = request.params.size() > 1 && !request.params[1].isNull() ? request.params[1].get_int() : -1;
-    int minConfs  = request.params.size() > 2 && !request.params[2].isNull() ? request.params[2].get_int() : -1;
-    int maxConfs  = request.params.size() > 3 && !request.params[3].isNull() ? request.params[3].get_int() : -1;
+    const int minBtcHeight = request.params.size() > 0 ? request.params[0].get_int() : -1;
+    const int maxBtcHeight = request.params.size() > 1 ? request.params[1].get_int() : -1;
+    const int minConfs = request.params.size() > 2 ? request.params[2].get_int() : -1;
+    const int maxConfs = request.params.size() > 3 ? request.params[3].get_int() : -1;
+    const int startBtcHeight = request.params.size() > 4 ? request.params[4].get_int() : -1;
```

Comment:
inputs are mentioned as `optional`. so someone might query with `startBtcHeight` and `limit` only.
Then the size based input argument parsing fails. Can we implement key based input argument parsing? (i.e !request.params['startBtcHeight'])